ApplicationLogs/LogReader.cs

```diff
@@ -39,5 +45,59 @@ public JObject OnProcess(HttpContext context, string method, JArray _params)
         public void PostProcess(HttpContext context, string method, JArray _params, JObject result)
         {
         }
+
+        public void OnPersist(Snapshot snapshot, IReadOnlyList<Blockchain.ApplicationExecuted> applicationExecutedList)
+        {
+            writeBatch = new WriteBatch();
+
+            foreach (var appExec in applicationExecutedList)
+            {
+                JObject json = new JObject();
+                json["txid"] = appExec.Transaction.Hash.ToString();
+                json["executions"] = appExec.ExecutionResults.Select(p =>
+                {
+                    JObject execution = new JObject();
+                    execution["trigger"] = p.Trigger;
+                    execution["contract"] = p.ScriptHash.ToString();
+                    execution["vmstate"] = p.VMState;
+                    execution["gas_consumed"] = p.GasConsumed.ToString();
+                    try
+                    {
+                        execution["stack"] = p.Stack.Select(q => q.ToParameter().ToJson()).ToArray();
+                    }
+                    catch (InvalidOperationException)
+                    {
+                        execution["stack"] = "error: recursive reference";
+                    }
+                    execution["notifications"] = p.Notifications.Select(q =>
+                    {
+                        JObject notification = new JObject();
+                        notification["contract"] = q.ScriptHash.ToString();
+                        try
+                        {
+                            notification["state"] = q.State.ToParameter().ToJson();
+                        }
+                        catch (InvalidOperationException)
+                        {
+                            notification["state"] = "error: recursive reference";
+                        }
+                        return notification;
+                    }).ToArray();
+                    return execution;
+                }).ToArray();
+                db.Put(WriteOptions.Default, appExec.Transaction.Hash.ToArray(), json.ToString());
+                writeBatch.Put(appExec.Transaction.Hash.ToArray(), json.ToString());
+            }
+        }
+
+        public void OnCommit(Snapshot snapshot)
+        {
+            db.Write(WriteOptions.Default, writeBatch);
```

Comment:
Why not call `db.Write()` in `OnPersist()`?

Reply:
Because `OnCommit` is where things are supposed to be committed to a database, right?

Reply:
I guess it is better to write the batch in `OnPersist()` though since the data should be idempotent if you write it twice.

Reply:
I moved it to `OnPersist()`